src/aws-cpp-sdk-core/source/http/curl/CurlHttpClient.cpp

```diff
@@ -436,16 +436,33 @@ static int CurlProgressCallback(void *userdata, double, double, double, double)
         curl_easy_pause(context->m_curlHandle, CURLPAUSE_CONT);
         return 0;
     }
-    // forcing "underflow" on the IOStream with ConcurrentStreamBuf to move data from back buffer to put area
-    int peekVal = ioStream->peek();
-    AWS_UNREFERENCED_PARAM(peekVal);
 
-    // forcing curl to try to ReadBody again (~to poll body IOStream for HTTP2)
-    // This is a spin pause-unpause in case of no data provided by a customer callback
-    // But otherwise curl will slow down the transfer and start calling as at frequency of 1s
-    //   see https://curl.se/mail/lib-2020-07/0046.html
-    // we should use multi handle or another HTTP client in the future to avoid this
-    curl_easy_pause(context->m_curlHandle, CURLPAUSE_CONT);
+    if (context->m_client->m_perfMode == TransferLibPerformanceMode::LOW_LATENCY)
+    {
+        // forcing "underflow" on the IOStream with ConcurrentStreamBuf to move data from back buffer to put area
+        int peekVal = ioStream->peek();
+        AWS_UNREFERENCED_PARAM(peekVal);
+
+        // forcing curl to try to ReadBody again (~to poll body IOStream for HTTP2)
+        // This is a spin pause-unpause in case of no data provided by a customer callback
+        // But otherwise curl will slow down the transfer and start calling as at frequency of 1s
+        //   see https://curl.se/mail/lib-2020-07/0046.html
+        // we should use multi handle or another HTTP client in the future to avoid this
+        curl_easy_pause(context->m_curlHandle, CURLPAUSE_CONT);
+    }
+    else
+    {
+        char output[1];
+        if (ioStream->readsome(output, 1) > 0)
```

Comment:
could we use [peek](https://en.cppreference.com/w/cpp/io/basic_istream/peek)? that way we dont have to do any unwinding/rewinding?

Reply:
....because of my hacks to make custom IOStream work as an async non-closable stream
peak will always return something ('a' or 'z'), while readsome will return only if there is something to read.